packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx

```diff
@@ -0,0 +1,688 @@
+        render(<PaymentMethodTitleTest {...defaultProps} />);
+
+        expect(screen.getByRole('heading', { name: defaultProps.method.config.displayName })).toBeInTheDocument();
+    });
+
+    it('renders name of payment method if it is credit card method', () => {
+        render(
+            <PaymentMethodTitleTest
+                {...defaultProps}
+                method={{
+                    ...defaultProps.method,
+                    method: 'credit-card',
+                }}
+            />,
+        );
+
+        expect(screen.getByRole('heading', { name: defaultProps.method.config.displayName })).toBeInTheDocument();
+    });
+
+    it('renders logo based on their method type', () => {
+        render(
+            <PaymentMethodTitleTest
+                {...defaultProps}
+                method={{
+                    ...defaultProps.method,
+                    method: PaymentMethodType.ApplePay,
+                }}
+            />,
+        );
+
+        const expectedPath = /^http/.test(LOGO_PATHS[PaymentMethodType.ApplePay])
+            ? LOGO_PATHS[PaymentMethodType.ApplePay]
+            : `${config.cdnPath}${LOGO_PATHS[PaymentMethodType.ApplePay]}`;
+
+        expect(screen.getByRole('img')).toHaveAttribute('src',  expectedPath);
+    });
+
+    it('renders logo for based on their method id', () => {
+        render(
+            <PaymentMethodTitleTest
+                {...defaultProps}
+                method={{
+                    ...defaultProps.method,
+                    id: PaymentMethodId.AmazonPay,
+                    method: 'widget',
+                }}
+            />,
+        );
+
+        expect(screen.getByRole('img')).toHaveAttribute('src', `${config.cdnPath}${LOGO_PATHS[PaymentMethodId.AmazonPay]}`);
+    });
+
+    it('renders logo based on their gateway id', () => {
+        render(
+            <PaymentMethodTitleTest
+                {...defaultProps}
+                method={{
+                    ...defaultProps.method,
+                    gateway: 'afterpay',
+                    method: 'multi-option',
+                }}
+            />,
+        );
+
+        expect(screen.getByRole('img')).toHaveAttribute('src', `${config.cdnPath}${LOGO_PATHS.afterpay}`);
+    });
+    describe('renders both logo and name for some hosted payment methods', () => {
+        it('renders both logo and name for Affirm hosted payment methods', () => {
+            render(
+                <PaymentMethodTitleTest
+                    {...defaultProps}
+                    method={{
+                        ...defaultProps.method,
+                        id: PaymentMethodId.Affirm,
+                    }}
+                />,
+            );
+    
+            const logos = screen.getByRole('img');
+            const names = screen.getByRole('heading');
+    
+            expect(logos).toBeInTheDocument();
+    
+            expect(names).toBeInTheDocument();
```

Comment:
Can we make the selection more method specific?
Do we know the img url or heading text here?

Reply:
there is no need to do it, because our component is simple and it has just title and logo. So we don't have a need to test logo src or text content here. Just need to test is there both text and logo or just logo.

Reply:
I believe we can all agree that the `checkout-js` is constantly changing.
As a result, we should make certain that the logo and name continuously satisfy our expectations in the long run.